src/postprocessors.rs

```diff
@@ -16,3 +17,32 @@ pub fn softbreaks_to_hardbreaks(
     }
     PostprocessorResult::Continue
 }
+
+pub fn filter_by_tags(
+    skip_tags: Vec<String>,
+    only_tags: Vec<String>,
+) -> impl Fn(&mut Context, &mut MarkdownEvents) -> PostprocessorResult {
+    move |context: &mut Context, _events: &mut MarkdownEvents| -> PostprocessorResult {
+        if !skip_tags.is_empty() || !only_tags.is_empty() {
+            match context.frontmatter.get("tags") {
+                Some(Value::Sequence(tags)) => {
+                    let skip = skip_tags
+                        .iter()
+                        .any(|tag| tags.contains(&Value::String(tag.to_string())));
+                    let include = only_tags.is_empty()
+                        || only_tags
+                            .iter()
+                            .any(|tag| tags.contains(&Value::String(tag.to_string())));
+                    if skip || !include {
+                        PostprocessorResult::StopAndSkipNote
+                    } else {
+                        PostprocessorResult::Continue
+                    }
+                }
+                _ => PostprocessorResult::Continue,
```

Comment:
Thinking through this a little more, I don't think this case is always correct.

When I run with `--only-tags publish` (that is, `skip_tags = []`, `only_tags = ["publish"]`), I would expect notes without frontmatter (or frontmatter but no `tags` value) to be skipped (`PostprocessorResult::StopAndSkipNote`).

In other words, no frontmatter or a missing `tags` attribute should I think be treated as if `tags` was present, but empty.

Reply:
This postprocessor probably also benefits from some unit tests to verify these various invariants, instead of relying only on the integration tests.

Do you think you could add some? Would you need some extra pointers on how to go about this?

Reply:
i think i have some time tomorrow to finish this

Reply:
FINALLY I found time again - and (hopefully) fixed up the last bits.
Let me know if this is now the logic you're expecting.

Reply:
mh-trimble == epsilonhalbe, one is a work account the other my personal